storage-proofs/src/zigzag_graph.rs

```diff
@@ -359,31 +390,56 @@ where
     // we would only need to compute the parents in one direction and with
     // that fill both caches.
     #[inline]
-    fn expanded_parents(&self, node: usize) -> Vec<usize> {
-        if let Some(parents) = self.read_parents_cache(node) {
-            return parents;
-        }
-
-        let parents: Vec<usize> = (0..self.expansion_degree)
-            .filter_map(|i| {
-                let other = self.correspondent(node, i);
-                if self.reversed {
-                    if other > node {
-                        Some(other)
+    fn expanded_parents<F, T>(&self, node: usize, mut cb: F) -> T
```

Comment:
This method (and `Graph.parents()`) are pretty hard for me to read. I'm not saying change anything in this PR, but in the future we may want to document them a little bit more.

Reply:
improved the documentation

Reply:
nice, thanks